### PR TITLE
Refactor `JournalMaster` to extend `Master` class

### PR DESCRIPTION
**Pull request by jenoudet**
### What changes are proposed in this pull request?
Refactor the `JournalMaster` class to behave like another `Master` class that can report its own gRPC services.

### Why are the changes needed?
- It addresses the `TODO(ggezer) Merge this with registerServices() logic.` [here](https://github.com/Alluxio/alluxio/blob/8183d1b976eaa0423014fb0d20985084029416f4/core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java#L576).
- It allows all gRPC services to be accessed in a unified manner: `mRegistry.getServers()` and then `server.getServices()` returns all the necessary services to be registered. This in turn allows nimbler manipulation of the gRPC services and gRPC servers as a whole, as they all the services can be accessed in the same way. 

### Does this PR introduce any user facing changes?
No.

### CODE REVIEW

**Comment by yyongycy** [comment]
Wondering whether this change impacts on the main logic and what test has been done?

File: core/server/common/src/main/java/alluxio/master/journal/noop/NoopPrimarySelector.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import alluxio.master.PrimarySelector;
+import alluxio.util.interfaces.Scoped;
+
+import java.net.InetSocketAddress;
+import java.util.function.Consumer;
+
+/**
+ * Noop primary selector.
+ */
+public class NoopPrimarySelector implements PrimarySelector {
```
Comment (elega) [inline core/server/common/src/main/java/alluxio/master/journal/noop/NoopPrimarySelector.java:24]: do you want to consolidate this with https://github.com/Alluxio/alluxio/blob/master/core/server/master/src/test/java/alluxio/master/AlwaysStandbyPrimarySelector.java?

File: core/common/src/main/java/alluxio/Constants.java [hunk 2]
```diff
@@ -120,7 +121,7 @@ public final class Constants {
   public static final String METRICS_MASTER_CLIENT_SERVICE_NAME = "MetricsMasterClient";
   public static final String BLOCK_WORKER_CLIENT_SERVICE_NAME = "BlockWorkerClient";
   public static final String FILE_SYSTEM_WORKER_CLIENT_SERVICE_NAME = "FileSystemWorkerClient";
-  public static final String JOURNAL_MASTER_CLIENT_SERVICE_NAME = "JournalMaster";
```
Comment (elega) [inline core/common/src/main/java/alluxio/Constants.java:123]: will this change cause any backward compatibility issues? e.g. in HA mode doing a rolling upgrade, or using old client connecting to the new masters.

**Review by elega — COMMENTED** [review]
lgtm overall. left two minor comments.
thanks for the refactoring

**Comment by yyongycy** [comment]
What if we push the journal down in the future?

File: core/server/common/src/main/java/alluxio/master/journal/noop/NoopPrimarySelector.java (hunk 1 above)
Comment (jenoudet) [reply core/server/common/src/main/java/alluxio/master/journal/noop/NoopPrimarySelector.java:24]: Great idea, I will refactor to use only the existing `AlwaysStandbyPrimarySelector`.

File: core/common/src/main/java/alluxio/Constants.java (hunk 2 above)
Comment (jenoudet) [reply core/common/src/main/java/alluxio/Constants.java:123]: When investigating its usages, I only saw it being used in log messages or error messages. I could not find a usage that affected the behavior of the client or server. Let me know if you find such a case.

**Comment by jenoudet** [comment]
> Wondering whether this change impacts on the main logic and what test has been done?

The `JournalMaster` ensures the functionality of the `fsadmin journal quorum` subcommands. The tests for its correct behavior can be found in `QuorumCommandIntegrationTest`. Given that these pass, I figured that the refactor does not impact the main logic.

**Review by yyongycy — APPROVED** [review]
LGTM

**Review by elega — APPROVED** [review]
LGTM

**Review by tcrain — APPROVED** [review]

**Comment by Xenorith** [comment]
alluxio-bot, merge this please

**Pull request merged**